pkg/cmd/cli/cmd/whoami.go

```diff
@@ -50,3 +54,39 @@ func (o WhoAmIOptions) WhoAmI() (*userapi.User, error) {
 
 	return me, err
 }
+
+func RunWhoAmI(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []string, o *WhoAmIOptions) error {
+	if kcmdutil.GetFlagBool(cmd, "token") {
+		cfg, err := f.OpenShiftClientConfig.ClientConfig()
+		if err != nil {
+			return err
+		}
+		if len(cfg.BearerToken) == 0 {
+			return fmt.Errorf("no token is currently in use for this session")
+		}
+		fmt.Fprintf(out, "%s\n", cfg.BearerToken)
```

Comment:
No, it's that the point of this command is not to validate the token.  It's to make it easy for you to get the token from something you already have stored locally.  If it's invalid you're going to find out when you make your call, and that's intentional.

----- Original Message -----
> > @@ -50,3 +54,39 @@ func (o WhoAmIOptions) WhoAmI() (*userapi.User, error) {
> >  
> >  	return me, err
> >  }
> > +
> > +func RunWhoAmI(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command,
> > args []string, o *WhoAmIOptions) error {
> > +	if kcmdutil.GetFlagBool(cmd, "token") {
> > +		cfg, err := f.OpenShiftClientConfig.ClientConfig()
> > +		if err != nil {
> > +			return err
> > +		}
> > +		if len(cfg.BearerToken) == 0 {
> > +			return fmt.Errorf("no token is currently in use for this session")
> > +		}
> > +		fmt.Fprintf(out, "%s\n", cfg.BearerToken)
> 
> > Why does it need to determine if it's valid? If you're getting a token,
> > you're going to use the token. If you wanted a valid token, i think you'd
> > want to use a different path.
> 
> What would drive a user to ever wanting an invalid token?  If you have it,
> you want to use it.  If you want to use it, it needs to be valid.  So a
> command that says, "here's an invalid token" is not very useful.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/3055/files#r32141280